src/components/atomic-swap/exchange.tsx

```diff
@@ -154,17 +161,30 @@ export const Exchange = (props: ExchangeProps) => {
           ) as Transaction<Memo<MemoType>, Operation[]>;
 
           const txSim = await server.simulateTransaction(tx);
+
           const preparedTransaction = assembleTransaction(
             tx,
             props.networkDetails.networkPassphrase,
             txSim,
-            xdr.LedgerFootprint.fromXDR(
-              Buffer.from(originalFootprint, "base64"),
-            ),
           );
 
+          const originalFootprintXDR = xdr.LedgerFootprint.fromXDR(
+            Buffer.from(originalFootprint, "base64"),
+          );
```

Comment:
you keep forgetting that fromXDR can do it from b64 directly :smile: 
```suggestion
          const originalFootprintXDR = xdr.LedgerFootprint.fromXDR(originalFootprint, "base64");
```

Reply:
ah lol, I'll only need you to remind me 10 more times don't worry 😝

Reply:
done in d67255ea7ce51c9f9101743ea6ba76a86c226d14